components/brave_new_tab_ui/components/default/cryptoDotCom/style.ts

```diff
@@ -0,0 +1,363 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License. v. 2.0. If a copy of the MPL was not distributed with this file.
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import styled from 'styled-components'
+interface StyleProps {
```

Comment:
I'm not going to comment as to my opinion on this style of mixing everything together in JS 😛 , but can we both remove props and code that isn't being used (some of these style props aren't), and consolidate all the duplicated code and put in to a function (e.g. `getCommonStyle(p)`)?

Reply:
We can definitely remove props/components that aren't being used. My bad for not doing that already.

My personal view on this is that if we're going to continue using Styled Components, we should have a base component that defines atomic, utility props (like `p={20}`) from which all other styled components extend. This would be super easy with `withComponent`, but that appears to be on the [deprecation block](https://styled-components.com/docs/api#withcomponent).

Reply:
I don't think this is a task for this PR, but here's an example of what I'm referring to with the base component idea: https://codesandbox.io/s/white-flower-5mikc?file=/src/style.js

cc. @ryanml

Reply:
With this pattern, any commonly used props can simply be lifted into the base component, making the other styled components less cluttered. Combined with a well thought out component library (that will translate across widgets), I think this will give us reusable components that are flexible enough to handle one-off cases in widgets without having to create a brand new style component, or a bunch of new props for minor styling differences.

Reply:
Well I personally wouldn't use atomic css for quite a few reasons at the moment so I don't think we need to change all the global components just yet. But you're welcome to create them for this or all widgets of course at least at first. We use base components already in many places, so of course we have the concept. But for this I simply meant to avoid repetition for every component, make a function: 

```
function getBoxStyle (p: StyleProps) {
  ${(p) =>
    p.$p &&
    `padding: ${typeof p.$p === 'number' ? `${p.$p}px` : p.$p}`}
  ${(p) =>
    p.$pt &&
    `padding-top: ${typeof p.$pt === 'number' ? `${p.$pt}px` : p.$pt}`}
  ${(p) =>
    p.$pr &&
    `padding-right: ${typeof p.$pr === 'number' ? `${p.$pr}px` : p.$pr}`}
  ${(p) =>
    p.$pb &&
    `padding-bottom: ${typeof p.$pb === 'number' ? `${p.$pb}px` : p.$pb}`}
  ${(p) =>
    p.$pl &&
    `padding-left: ${typeof p.$pl === 'number' ? `${p.$pl}px` : p.$pl}`}
  ${(p) =>
    p.$m &&
    `margin: ${typeof p.$m === 'number' ? `${p.$m}px` : p.$m}`}
  ${(p) =>
    p.$mt &&
    `margin-top: ${typeof p.$mt === 'number' ? `${p.$mt}px` : p.$mt}`}
  ${(p) =>
    p.$mr &&
    `margin-right: ${typeof p.$mr === 'number' ? `${p.$mr}px` : p.$mr}`}
  ${(p) =>
    p.$mb &&
    `margin-bottom: ${typeof p.$mb === 'number' ? `${p.$mb}px` : p.$mb}`}
  ${(p) =>
    p.$ml &&
    `margin-left: ${typeof p.$ml === 'number' ? `${p.$ml}px` : p.$ml}`}
}

const aList = styled<{StyleProps & ListProps>('ul')`
  ${getBoxStyle}
`
```